### PR TITLE
Migrate runtime propensity scripts to TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**

### CODE REVIEW

File: src/runtime/propensity-server.ts [hunk 1]
```diff
@@ -18,74 +18,78 @@ import {
...
+  constructor(
+    private readonly win_: Window,
+    private readonly deps_: Deps,
+    private readonly fetcher_: Fetcher
+  ) {
     this.publicationId_ = this.deps_.pageConfig().getPublicationId();
-    /** @private {?string} */
-    this.clientId_ = null;
-    /** @private @const {!./fetcher.Fetcher} */
-    this.fetcher_ = fetcher;
```
Comment (justinchou-google) [inline src/runtime/propensity-server.ts:48]: Does `this.fetcher_` need to be set here?

File: src/runtime/propensity.ts [hunk 2]
```diff
@@ -69,22 +71,20 @@ export class Propensity {
...
+  getPropensity(type?: PropensityType): Promise<PropensityScore | null> {
+    if (type && !Object.values(PropensityType).includes(type)) {
       throw new Error('Invalid propensity type requested');
     }
     if (!type) {
-      type = PropensityApi.PropensityType.GENERAL;
+      type = PropensityType.GENERAL;
     }
     return this.propensityServer_.getPropensity(
       this.win_.document.referrer,
```
Comment (justinchou-google) [inline src/runtime/propensity.ts:82]: Similarly, I think this requires `this._win` to be set?

File: src/runtime/propensity-server.ts (hunk 1 above)
Comment (ChrisAntaki) [reply src/runtime/propensity-server.ts:48]: Oh, the new code uses (`parameter properties`)[https://google.github.io/styleguide/tsguide.html#parameter-property-comments] to simplify the code. So `this.fetcher_` basically gets set on line 63 now

File: src/runtime/propensity.ts (hunk 2 above)
Comment (ChrisAntaki) [reply src/runtime/propensity.ts:82]: Yep, it's a `parameter property`

**Review by justinchou-google — APPROVED** [review]

**Pull request merged**